src/components/editor-panels.js

```diff
@@ -2,30 +2,40 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Spinner, CheckboxControl, PanelBody, PanelRow } from '@wordpress/components';
+import { BaseControl, CheckboxControl, PanelBody, Spinner } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
 
-const CheckboxesGroup = ( { options, values, onChange } ) => {
+const CheckboxesGroup = ( { options, values, onChange, defaultRequiredSlug } ) => {
 	if ( ! Array.isArray( options ) ) {
 		return <Spinner />;
 	}
-	return options.map( ( { name, slug } ) => (
-		<PanelRow key={ slug }>
+	return options.map( ( { name, slug } ) => {
+		const isDefaultType = slug === defaultRequiredSlug;
+		const otherTypesSelected = values.length > 1 || ( values.length === 1 && values[0] !== defaultRequiredSlug );
+		const isDisabled = isDefaultType && !otherTypesSelected;
```

Comment:
Nit: explicitly account for `defaultRequiredSlug` prop not being passed, and simplify check for other selected options:

```suggestion
		const isDefault = defaultRequiredSlug && slug === defaultRequiredSlug;
		const otherOptionsSelected = defaultRequiredSlug && values.some( value => value !== defaultRequiredSlug );
		const isDisabled = isDefault && ! otherOptionsSelected;
```